### ISSUE TITLE
Failed to download file from URL https://github.com/avocado-framework/avocado/archive/74.0.22.g06ec7688.tar.gz#/avocado-74.0.22.g06ec7688.tar.gz

### ISSUE THREAD

**Issue by jpopelka** [packit/packit-service#313]
- https://sentry.io/organizations/red-hat-0p/issues/1421880069
- https://github.com/avocado-framework/avocado/pull/3447

**Comment by lachmanfrantisek** [comment]
> 2

Rebase helper probably don't understand this part of the specfile so we don't update the Source:

```
%if 0%{?rel_build}
Source0: https://github.com/avocado-framework/%{srcname}/archive/%{version}.tar.gz#/%{gittar}
%else
Source0: https://github.com/avocado-framework/%{srcname}/archive/%{commit}.tar.gz#/%{gittar}
%endif
```

**Comment by dhodovsk** [comment]
Here is some green pass of packit on this repo - https://github.com/avocado-framework/avocado/pull/3424 that's weird? were there some rebase-helper updates since?

**Comment by lachmanfrantisek** [comment]
> were there some rebase-helper updates since?

There was a change on our site -- we are downloading the remote sources since that... (What cause that problem.) It looks like it did not use the PR code, so it was green but with the wrong SRPM.)

**Comment by TomasTomecek** [comment]
would be nice to
1) report to rebase-helper
2) get in touch with people from avocado and inform them about this problem on our side

**Comment by TomasTomecek** [comment]
```
%if 0%{?rel_build}
%global archive_suffix %{version}.tar.gz#/%{gittar}
%else
%global archive_suffix %{commit}.tar.gz#/%{gittar}
%endif

Source0: https://github.com/avocado-framework/%{srcname}/archive/%archive_suffix
```

**Comment by dhodovsk** [comment]
Can be seen also here: https://github.com/dhodovsk/cython/pull/1

Spec:
https://github.com/dhodovsk/cython/blob/da54e71eca298258cd3f37bcadfdf4ea7d23ba8d/Cython.spec#L14


Error message:
```
Preparing of the upstream to the SRPM build failed: Failed to download file from URL https://github.com/cython/cython/archive/0.29.14.589.gda54e71ec/Cython-0.29.14.589.gda54e71ec.tar.gz. Reason: 'Not Found'. 
```

**Comment by dhodovsk** [comment]
Fixed here: https://github.com/avocado-framework/avocado/pull/3554

The issue was:
- packit uses upstream_package_name for setting the name of tar.gz archive `(avocado-framework-{version})`
- the spec expected different name in source `avocado-(version)`

In which places is `upstream_package_name` used? Could we introduce other option `archive_name` that will be used while running default `fix-spec` and `create-archive` actions?

**Comment by TomasTomecek** [comment]
@hodovska +1 archive name

**Comment by TomasTomecek** [comment]
What's the update here?

**Comment by lachmanfrantisek** [comment]
> Fixed here: avocado-framework/avocado#3554
> The issue was:

Looks like it's already fixed. Please, reopen if you know about any missing peace...
